packages/palette/src/elements/DonutChart/DonutChart.tsx

```diff
@@ -0,0 +1,164 @@
+import { interpolate } from "d3-interpolate"
+import { arc as d3_arc, pie as d3_pie } from "d3-shape"
+import React, { useRef, useState } from "react"
+import { Spring } from "react-spring"
+import styled from "styled-components"
+import { color, space } from "../../helpers"
+import { Color } from "../../Theme"
+import { Box } from "../Box"
+import { ChartHoverTooltip } from "../DataVis/ChartHoverTooltip"
+import { coerceTooltip } from "../DataVis/ChartTooltip"
+import { ProvideMousePosition } from "../DataVis/MousePositionContext"
+import { ChartProps } from "../DataVis/utils/SharedTypes"
+import { useGetWrapperWidth } from "../DataVis/utils/useGetWrapperWidth"
+import { useHasEnteredViewport } from "../DataVis/utils/useHasEnteredViewPort"
+import { Sans } from "../Typography"
+
+const colors: Color[] = ["black10", "black30", "black60"]
+
+export interface DonutChartProps extends ChartProps {
+  margin?: number
+}
+
+/**
+ * DonutChart is a component that displays data points with donut shaped arcs.
+ * Good for illustrating numerical proportions.
+ */
+export const DonutChart: React.FC<DonutChartProps> = ({
+  points,
+  margin = space(3),
+}: DonutChartProps) => {
+  const [hoverIndex, setHoverIndex] = useState(-1)
+  const [rest, setRest] = useState(false)
+
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  const hasEnteredViewport = useHasEnteredViewport(wrapperRef)
+
+  const width = useGetWrapperWidth(wrapperRef)
+
+  const donutWidth = width * 0.12
+  const donutLabelMargin = space(2) / Math.sqrt(2)
```

Comment:
why the division by root 2 here?

Reply:
Good catch 🙌  it is not needed anymore since the `donutLabelMargin` is added to the radius of `labelArc` and `d3` takes care of the angle.

Previously I was calculating `x, y` distances manually and added this, which also was wrong, I should have had `sin`/`cos`.